### ISSUE TITLE
Fraction_input Parameter Semantics: selected nodes are excluded from input, not connected to input

### ISSUE THREAD

**Issue by Fixstern0211** [Cyber-Physical-Systems-in-Mech-Eng/pyReCo#103]
**Issue:**
In custom_models.py, the method _connect_input_to_reservoir() (line 692–722) selects nodes based on fraction_input, then zeros out their input weights:

_input_receiving_nodes = node_selector.select_nodes(
      fraction=self.reservoir_layer.fraction_input
  )
node_mask = np.ones_like(full_input_weights)
node_mask[input_receiving_nodes] = 0
self._set_readin_weights(weights=(full_input_weights * node_mask))_

This means fraction_input=0.1 results in 90% of nodes receiving input (10% excluded), not 10% receiving input. The variable name
input_receiving_nodes and the comment "select the fraction of nodes that are input nodes" suggest the opposite behavior.

**Verification**
from pyreco.models import ReservoirComputer as RC
import numpy as np

for fi in [0.1, 0.3, 0.5, 0.9]:
    np.random.seed(42)
    model = RC(num_nodes=100, density=0.01, fraction_input=fi, spec_rad=0.9, leakage_rate=0.8, fraction_output=1.0, optimizer='ridge')
    model.fit(np.random.randn(20, 100, 3), np.random.randn(20, 1, 3))
    W_in = model.model.input_layer.weights
    active = np.sum(np.any(W_in != 0, axis=1))
    print(f"fraction_input={fi}: {active}/100 nodes receive input ({active}%)")

Output:
  fraction_input=0.1: 90/100 nodes receive input (90%)
  fraction_input=0.3: 70/100 nodes receive input (70%)
  fraction_input=0.5: 50/100 nodes receive input (50%)
  fraction_input=0.9: 10/100 nodes receive input (10%)

**Impact**
Users relying on fraction_input to control the number of input-connected nodes will get the inverse of what they expect. 

**Suggested fix**
1. Rename the parameter to clarify its meaning (e.g., fraction_input_excluded)
2. Or invert the mask logic so that selected nodes keep their weights instead of losing them

CC: @mertstend  @maneesh51
